src/humanize/time.py

```diff
@@ -223,6 +224,47 @@ def naturaltime(value, future=False, months=True, minimum_unit="seconds", when=N
     return ago % delta
 
 
+def precisetime(
+    value, future=False, minimum_unit="seconds", suppress=(), format="%0.2f", when=None
+):
+    """Precise like `precisedelta` but includes tense like `naturaltime`.
+
+    Args:
+        value (datetime.datetime): A `datetime` or a number of seconds.
+        future (bool): Ignored for `datetime`s, where the tense is always figured out
+            based on the current time. For integers, the return value will be past tense
+            by default, unless future is `True`.
+        minimum_unit (str): The lowest unit that can be used.
+        suppress: If desired, some units can be suppressed:
+        you will not see them represented and the time of the other
+            units will be adjusted to keep representing the same timedelta
+        when (datetime.datetime): Point in time relative to which _value_ is
+            interpreted.  Defaults to the current time in the local timezone.
+
+    Returns:
+        str: A precise representation of the input in a resolution that makes sense.
+    """
+    now = when or _now()
+    date, delta = date_and_delta(value, now=now)
+    if date is None:
+        return value
+    # determine tense by value only if datetime/timedelta were passed
+    if isinstance(value, (dt.datetime, dt.timedelta)):
+        future = date > now
+
+    ago = _("%s from now") if future else _("%s ago")
+    delta = precisedelta(
+        delta, minimum_unit, suppress=suppress, format=format, when=now
+    )
+
+    # janky but probably relatively efficient to check for "0 [any unit]"
+    splitdelta = delta.split(" ")
+    if len(splitdelta) == 2 and float(splitdelta[0]) == 0:
```

Comment:
Will this break for languages where the unit is two or more words?

For example: 

https://github.com/jmoiron/humanize/blob/99b68f6564435ea9b300cdb4a9b86153d31a6ab5/src/humanize/locale/id_ID/LC_MESSAGES/humanize.po#L171-L181

Reply:
ah, wasn't aware that was a case
are there any languages that dont have a space between the number and the word? if not I have an idea to work with this

Reply:
I think they all have a space between the number and the word, at least for the `*seconds` I checked.